src/components/Friends.js

```diff
@@ -0,0 +1,89 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux'
+import {getAllUsers, followFrined, getUser} from '../redux/users';
+import {Link} from 'react-router-dom'
+
+
+
+class Friends extends Component {
+    constructor () {
+      super()
+      this.state = {
+        username:'',
+        userDetail:null,
+        
+      }
+      this.addFriend=this.addFriend.bind(this)
+      this.handleSearch=this.handleSearch.bind(this)
+      this.handleSubmit=this.handleSubmit.bind(this)
+    }
+     componentDidMount(){
+     this.props.getAllUsers()
+      
+    }
+  
+    handleSearch (e) {
+      this.setState({ username: e.target.value })
+    }
+  
+    handleSubmit (e) {
+      e.preventDefault()
+
+    }
+
+    addFriend(user,friendId){
+      
+      if(user.friends === null){
+        const arr=[friendId]
+        this.props.followFrined(user.id, arr)
+      }else{ 
+        const arr=[...user.friends ,friendId]
+        this.props.followFrined(user.id, arr)
+      }
+   
+    }
+
+    render () {
+      const{handleSearch,handleSubmit}=this
+      const {username} = this.state
+      const {users}= this.props
+      return (
+      
+        <div className='searchbar-container'>
+          <form onSubmit={handleSubmit}>
+            <input type='text' size='45' placeholder='name' onChange={handleSearch} value={username} />
+            <button type='submit' onClick={handleSubmit}> Search</button>
+          </form>
+          <ul id="myUL" >
+                {
+                users.map( user => {
+                  if (user.first_name.toLowerCase().indexOf(username.toLowerCase()) > -1 && username!== '') {
```

Comment:
break up your code a little bit, into variables to make it more readable

instead of x.indexOf(y) > -1, x.includes(y) works.

instead of username !== '', you can make the input "required"